funnel/views/project.py

```diff
@@ -111,21 +114,43 @@ def project_has_tickets(obj):
     )
 
 
-@Project.features('has_tickets_or_rsvp')
-def project_has_tickets_or_rsvp(obj):
-    return obj.features.has_tickets() or obj.features.has_rsvp()
+@Project.features('tickets_or_rsvp')
+def feature_project_tickets_or_rsvp(obj):
+    return obj.features.tickets() or obj.features.rsvp()
+
+
+@Project.features('rsvp_register')
+def feature_project_register(obj):
+    rsvp = obj.rsvp_for(current_auth.user)
+    return rsvp is None or not rsvp.state.YES
+
+
+@Project.features('rsvp_deregister')
+def feature_project_deregister(obj):
+    rsvp = obj.rsvp_for(current_auth.user)
+    return rsvp is not None and rsvp.state.YES
 
 
 @Project.features('schedule_no_sessions')
-def project_has_no_sessions(obj):
+def feature_project_has_no_sessions(obj):
     return obj.schedule_state.PUBLISHED and not obj.schedule_start_at
 
 
 @Project.features('comment_new')
-def project_comment_new(obj):
+def feature_project_comment_new(obj):
     return obj.current_roles.participant is True
 
 
+@Project.features('registration')
+def feature_project_registration(obj):
+    return obj.rsvp_for(current_auth.user) is None
+
+
+@Project.features('deregistration')
+def feature_project_deregistration(obj):
+    return obj.rsvp_for(current_auth.user) is not None
```

Comment:
Should be `rsvp_unregistered` and `rsvp_registered`

Reply:
English convention seems to be "unregistered" for the state and "deregister" for the act. We could keep that, or unify into "deregistered" and "deregister". "Deregistered" by a strict English reading implies a past registration.